File: Code Samples/Queue.java
import java.util.Arrays;

class Queue<Type> {
	public int size = 0;
	private Object items[];
	private static final int DEFAULT_CAPACITY = 10;

	public Queue(){
		items = new Object[DEFAULT_CAPACITY];
	}
	private void resize(){
		int newSize = items.length * 2;
		items = Arrays.copyOf(items, newSize);
	}
	public void enqueue(Type item){
		if(size == items.length){
			resize();
		}
		items[size++] = item;
	}
	public Type dequeue() {
		Type item = (Type) items[--size];
		items[size] = null;
		return item;
	}
	public Boolean isEmpty(){
		return size == 0;
	}



	/* TESTS */
	public static void main(String[] args) {
		Queue s = new Queue<Integer>();

		s.push(4);
		s.push(5);

		assert !s.isEmpty();
		assert s.pop() == 5;
		assert s.pop() == 4;
		assert s.isEmpty();
	}
}

File: Code Samples/queue.py
class Queue(object) :
	def __init__(self):
		self.head = 0
		self.tail = 0
		self.items = []

	def __getitem__(self,key):
		return self.items[key]

	def enqueue(self, item):
		self.tail += 1
		self.items.append(item)

	def dequeue(self):
		self.head += 1
		return self.items.pop(0)

	def isEmpty(self):
		return self.head == self.tail


# Tests
q = Queue()
q.enqueue(4)
q.enqueue(5)

assert q.items == [4,5]
assert not q.isEmpty()
assert q[1]    == 5
assert q.dequeue() == 4
assert q.dequeue() == 5
assert q.isEmpty()